### PR TITLE
Add function specific secrets support to action

### PR DESCRIPTION
**Pull request by pavelzubarev**
Any opinions, likes/dislikes are welcomed!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,8 +15,9 @@ That API-key should have CDF capabilities required to run the code within the Fu
 1. `cdf_project`: Name of your CDF project/tenant. Inferred from your API-keys. Will be validated with API-keys if provided
 2. `cdf_base_url`: Base url of your CDF tenant, defaults to _https://api.cognitedata.com_
 3. `function_file`: Name of the main function python file (defaults to `handler.py`)
-4. `schedule_file`: File location with the list of schedules to be applied. see file format below. (defaults to None ie no schedules)
-5. `remove_only`: Checks that specified function is removed with all it's schedules. Deployment logic is skipped
+4. `function_secrets`: Name of Github secrets that holds the base64 encoded json dictionary with secrets.
```
Comment (scottmelhop) [inline README.md:18]: Name of THE Github secret .....
So the json dictionary needs to be base64 encoded in the secret?

Comment (pavelzubarev) [reply README.md]: yup. you can store whatever and we don't need to think much about what is in there

File: src/index.py [hunk 2]
```diff
@@ -42,6 +42,7 @@ def setup_config() -> FunctionConfig:
             runtime_key=os.getenv("INPUT_CDF_RUNTIME_CREDENTIALS"),
             cdf_base_url=os.getenv("INPUT_CDF_BASE_URL"),
         ),
+        secret=os.getenv("INPUT_FUNCTION_SECRETS") or None,
```
Comment (ks93) [inline src/index.py:45]: why `or None`? `os.getenv("") `returns None by default

File: tests/conftest.py [hunk 3]
```diff
@@ -51,6 +51,7 @@ def valid_config(monkeypatch, loggedin_status):
                 runtime_key="FUNCTION_KEY",
                 cdf_base_url="https://api.cognitedata.com",
             ),
+            secret="eyJrZXkiOiJ2YWx1ZSJ9Cg==",
```
Comment (ks93) [inline tests/conftest.py:54]: `# this is a mock example of an invalid key!`

**Review by ks93 — APPROVED** [review]
👍

File: src/index.py (hunk 2 above)
Comment (haakonvt) [reply src/index.py:45]: if secret is an empty string, this would then return None

Comment (ks93) [reply src/index.py:45]: Ah..

Comment (haakonvt) [reply src/index.py:45]: is it necessary, no idea 😂

Comment (pavelzubarev) [reply src/index.py:45]: I don't want to rely on default behavior of GitHub (aka they suddenly put empty strings)

File: tests/conftest.py (hunk 3 above)
Comment (pavelzubarev) [reply tests/conftest.py:54]: it's a valid secret which can be decoded and parsed by json.

File: README.md [hunk 4]
```diff
@@ -12,11 +12,12 @@ That API-key should have following CDF capabilities: `Files:READ`, `Files:WRITE`
...
  
 #### Optional
-1. `cdf_project`: Name of your CDF project/tenant. Inferred from your API-keys. Will be validated with API-keys if provided
+1. `cdf_project`: The name of your CDF project/tenant. Inferred from your API-keys. Will be validated with API-keys if provided
 2. `cdf_base_url`: Base url of your CDF tenant, defaults to _https://api.cognitedata.com_
-3. `function_file`: Name of the main function python file (defaults to `handler.py`)
-4. `schedule_file`: File location with the list of schedules to be applied. see file format below. (defaults to None ie no schedules)
-5. `remove_only`: Checks that specified function is removed with all it's schedules. Deployment logic is skipped
+3. `function_file`: The name of file with main function (defaults to `handler.py`)
+4. `function_secrets`: The name of Github secrets that holds the base64 encoded json dictionary with secrets.
```
Comment (haakonvt) [inline README.md:18]: Let's add an example of how to do this below, then add a small sentence like "see an example of how to do this in section `<section>`"

File: action.yaml [hunk 5]
```diff
@@ -25,6 +25,9 @@ inputs:
     description: Name of function file inside function folder, for instance, handler.py.
     default: handler.py
     required: false
+  function_secrets:
+    description: Base64 encoded string with json payload. Will be passed to a function's secrets as is.
+    required: false
```
Comment (haakonvt) [inline action.yaml:30]: As is? It sounds like FunctionsAPI.create kwarg `secrets` accepts this

File: src/index.py (hunk 2 above)
Comment (ks93) [reply src/index.py:45]: Why would github ever change any behaviour? :trollface: (master)

**Review by haakonvt — COMMENTED** [review]
Simple and elegant solution! We only need to show an example of how to create it - and give an example (maybe) of when it could be useful (like, the customer wants alerts directly to slack/foo/bar)

File: src/index.py (hunk 2 above)
Comment (haakonvt) [reply src/index.py:45]: Add a test for empty string scenario btw ;) :trollface:

File: action.yaml (hunk 5 above)
Comment (pavelzubarev) [reply action.yaml:30]: yup. that's the case. we just put that dictionary there and it will be given to the function implementation as it was stored here. (already decoded and parsed into dict)

File: tests/conftest.py (hunk 3 above)
Comment (ks93) [reply tests/conftest.py:54]: > `# this is a mock example of an val key!`

**Review by ks93 — APPROVED** [review]
❤️
